packages/flutter/lib/src/material/app_bar.dart

```diff
@@ -777,11 +782,28 @@ class _AppBarState extends State<AppBar> {
       _scrollNotificationObserver!.removeListener(_handleScrollNotification);
       _scrollNotificationObserver = null;
     }
+    _notificationListenerStates.clear();
     super.dispose();
   }
 
   void _handleScrollNotification(ScrollNotification notification) {
     if (notification is ScrollUpdateNotification && widget.notificationPredicate(notification)) {
+      // The notification context of a ScrollNotification points to the RawGestureDetector
+      // of the Scrollable. We get the ScrollableState associated with this notification
+      // by looking up the tree.
+      final ScrollableState? notificationScrollableState = notification.context?.findAncestorStateOfType<ScrollableState>();
+
+      // Check if any scroll positions on the vertical axis have been updated.
+      // If so, update the scroll position for the associated ScrollableState.
+      if (notificationScrollableState is ScrollableState && notification.metrics.axis == Axis.vertical) {
```

Comment:
Using `is` is not permitted by the style guide: https://github.com/flutter/flutter/blob/master/docs/contributing/Style-guide-for-Flutter-repo.md#avoid-interleaving-multiple-concepts-together